clog/clog.go

```diff
@@ -0,0 +1,143 @@
+/*
+Package clog provides Conext with logging information.
+*/
+package clog
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+// unique type to prevent assignment.
+type clogContextKey struct{}
```

Comment:
**nit:** Might have a specific instance of it to avoid the blank "instantiation" as well
```suggestion
type clogContextKeyT struct{}
var clogContextKey = clogContextKeyT{}
```
then just use as `clogContextKey`.